theurgy.lic

```diff
@@ -2,185 +2,453 @@
+  def get_rituals
+    rituals = all_rituals
+    blacklist = @settings.theurgy_blacklist
+    rituals.reject! { |r| blacklist.include? r[:method].to_s }
+    whitelist = @settings.theurgy_whitelist
+    unless whitelist.empty?
+      rituals.select! { |r| whitelist.include? r[:method].to_s }
+    end
+    rituals
+  end
+  
+  def all_rituals
+    rituals = []
+    if @data['almsbox'] && @settings.tithe
+      rituals << {
+        method: :tithe,
+      }
+    end
+    if @hometown == 'Crossing'
+      if DRStats.circle >= 30
+        rituals << {
+          method: :study_wall,
+        }
+      end
+      rituals << {
+        method: :refectory,
+      }
+    end
+    if DRSkill.getrank('Mechanical Lore') > 140 && @immortal_aspect && @data['limb_foraging']
+      # TODO: allow other knife nouns 
+      rituals << {
+        method: :carve_bead,
+        items: [
+          { name: 'carving knife', any_container: true },
+          { name: 'shaper', any_container: true},
+          { name: 'prayer chain', any_container: true },
+        ],
+      }
+    end
+    if @data['gather_sirese'] && @data['plant_sirese']
+      rituals << {
+        method: :sirese_seed,
+      }
+    end
+    if @data['bath']
+      herbs = @data['herbs'] || ['sage', 'lavender']
+      rituals << {
+        method: :bathe,
+        items: herbs.map { |h| { name: h, shop: @data['theurgy_supplies'] } },
+      }
+    end
+    if @immortal_aspect
+      ritals << {
+        method: :meditate_bead,
+        items: [
+          { name: 'prayer chain', any_container: true },
+        ],
+      }
+    end
+    rituals << {
+      method: :pray_badge,
+    }
+    # TODO: prayer mat support
+    if @data['altar']
+      if DRStats.circle >= 10
+        rituals << {
+          method: :dance,
+        }
+      end
+      rituals << {
+        method: :incense,
+        items: [
+          { name: 'incense', shop: @data['theurgy_supplies'] },
+          { name: 'flint', shop: @data['flint_shop'] },
+        ]
+      }
+      rituals << {
+        method: :wine,
+        items: [
+          { name: 'wine', shop: @data['theurgy_supplies'] },
+        ],
+      }
+      rituals << {
+        method: :recite_prayer,
+      }
+      rituals << {
+        method: :clean_altar,
+      }
+    end
+    rituals << {
+      method: :embarass_myself,
+      items: [
+        {
+          name: 'golden parchment',
+          shop: @data['parchment_shop'],
+          purchase_method: :buy_parchment,
+        }
+      ],
+    }
+    
+    rituals
+  end
+  
+  def get_communes
+    communes = []
+    if DRStats.circle > 3 && DRSkill.getrank('Outdoorsmanship') > 20 && @data['dirt_foraging']
+      communes << {
+        name: :eluned,
+        method: :commune_eluned,
+      }
+    end
+    if DRStats.circle > 2
+      communes << {
+        name: :tamsine,
+        method: :commune_tamsine
+      }
+    end
+    if DRStats.circle > 60
+      communes << {
```

Comment:
Truffenyi needs to support GG for faster times.

Reply:
Is this a _need_ or a nice to have?

Reply:
Need. Spending money on things and wasting time on purchasing seems pointless if you don't need it. Put in a waggle_set for truffenyi and if it exists, snapcast it, if not, go buy supplies and do it the longer way.

Reply:
Also, getting the berries in Shard is beyond hard. It's a wandering salesman Forf that sells them, which is why GG is supported in Shard.

Reply:
This is accommodating the higher circle Clerics, and not making us all work as if we are circle 5.

Reply:
It's also a stupidly simple addition. I feel like he's done all this work, and the things I'm requesting are such minor additions.

Reply:
I agree this is a good idea.  Do you know how to check if the player has the GG spell?  Just add a setting, or is there some programmatic way to check?

Reply:
Not without spell or tryign to cast it and saving it as a UserVar. I think the best way would be to let ;buff support it, so if it exists, let that take care of everything. I hacked quick-theurgy for it, but if not, I would have used a waggle_set.

Reply:
There's no common method for parsing a spell book at the moment.. 

With that in mind, a couple of ideas i can offer are to make a waggle_set called `theurgy` or somesuch and let the user define GG among any other needed spells. 

Another option is to do a one-time check on the output of `spells` and set a UserVar for the character `UserVars.knows_gg = true`